### DISCUSSION TITLE
generating new cite key when importing to a group

### DISCUSSION THREAD

**Discussion by bobkummerfeld** [retorquere/zotero-better-bibtex#1859] [Q&A] [answered]
If I have an item in my full library and it has a pinned citation key, when I try to add the item to a group it generates a new cite key, even though I had the key pinned in my library.   I've looked for an option in the config to control this but can't find one.

Is there a way to prevent generation of a new cite key when adding to a group?

Thanks!

**Comment by retorquere** [comment]
Pinned keys are those that exist in the `extra` field; is the content of the `extra` field not transferred to the group library when it is copied?

**Reply by bobkummerfeld** [reply]
Yes. For example, the original has "Citation Key: McBroom2020" in the Extra field but when I drag it to a group it loses that content.

**Reply by retorquere** [reply]
I'd report that as a bug to Zotero. Doesn't seem right that the whole extra fields would be dropped on copy. If the extra field gets copied, the citekey regeneration will go away.

**Reply by retorquere** [reply]
I understood your comment to mean that all of the extra field was dropped?

**Comment by retorquere** [comment, marked as answer]
Continued in #1860.
